lib/ZnapZend/ZFS.pm

```diff
@@ -553,7 +557,75 @@ sub sendRecvSnapshots {
     return 1;
 }
 
+sub filterPropertyNames {
+    # This routine is a helper for getXproperties to allow easy passing of
+    # specific properties we are interested to `zfs get`, to optimize some
+    # code around that. As an input it can get either:
+    # * string that would be passed through as is (after some sanitization);
+    # * array of strings that would be concatenated into a comma-separated
+    #   string with our name-space propertyPrefix (e.g. "org.znapzend:...")
+    #   prepended if needed;
+    # * undef for defaulting to 'all'
+    # Returns a string safe to pass into `zfs get`
+    my $self = shift;
+    my $propnames = shift;
+
+    my $propertyPrefix = $self->propertyPrefix;
+
+    if (defined($propnames) && $propnames) {
```

Comment:
this does the same

```suggestion
    if ($propnames) {
```

Reply:
So the shorter form is true if:
* propnames is defined,
* AND propnames is not (empty string OR zero number (OR maybe string "0" too?))

In particular, if it is NOT defined, there is no dereferencing error to check for further values?

Reply:
Question aside, the proposed fix is merged